iron-overlay-manager.html

```diff
@@ -293,15 +293,28 @@
     },
 
     /**
+     * Returns the deepest overlay in the path.
+     * @param {Array<Element>} path
+     * @return {Element|undefined}
+     * @private
+     */
+    _overlayInPath: function(path) {
+      for (var i = 0; i < path.length; i++) {
+        if (path[i]._manager === this) {
+          return path[i];
+        }
+      }
+    },
+
+    /**
      * Ensures the click event is delegated to the right overlay.
      * @param {!Event} event
      * @private
      */
     _onCaptureClick: function(event) {
       var overlay = /** @type {?} */ (this.currentOverlay());
-      // Check if clicked outside of any overlay.
-      var target = /** @type {Element} */ (Polymer.dom(event).rootTarget);
-      if (overlay && this._overlayParent(target) !== overlay) {
```

Comment:
Arguably that check could be removed, and we could allow focus to be returned to the previous overlay. This would allow to also remove the call to `this._manager.focusOverlay();` in `iron-overlay-behavior`'s `_applyFocus`.
I'll do this in a separate PR though.